### ISSUE TITLE
Is there a way to know the style is declutter？

### ISSUE THREAD

**Issue by longyi-xw** [openlayers/openlayers#15650]
**Describe the bug**
I know this isn't a bug, but I might need help.

**Expected behavior**
When my feature requires a combination of text and line styles at the same time, the text is decluttered using overflow, but the lines still exist. I don't have a way to judge and hide the lines. Do you know of any other ways I can use this combination?

it looks like this now
![image](https://github.com/openlayers/openlayers/assets/54786937/ab909f4e-da68-49bc-a923-feceaf67741b)

**Comment by ahocevar** [comment]
Unfortunately he only way to determine that is to do hit detection for a known point on the declutter candidate.

**Comment by longyi-xw** [comment]
@ahocevar In fact, hitDetectionRenderer can only tell me the target that the mouse is hitting, but it cannot tell me that all points on the map are decluttered. Is there a way for me to re-implement the text declutter myself?

**Comment by ahocevar** [comment]
@longyi-xw Yes there is, you could use https://github.com/Geovation/labelgun.

**Comment by longyi-xw** [comment]
@ahocevar Thank you for your suggestion, but I use styleFunction. I use three styles combined to form the visible points. If I use this method, I need to additionally render the same part of the data, and the points and labels are stripped. Is there any performance issue? There will be some impact. I now use vectorImage and style caching. There are more than 5,000 data. It seems a bit difficult to double the rendering and control the memory within 400.
